Makefile

```diff
@@ -375,3 +375,30 @@ release-push-%:
 	$(MAKE) clean-bin
 	$(MAKE) ARCH=$* build
 	$(MAKE) ARCH=$* push
+
+# Dependency management via dep (https://github.com/golang/dep)
+PHONHY: check-dep get-dep verify-vendor test-dep
+check-dep:
+	@if [ -z "$$(which dep)" ]; then \
+		echo 'Missing `dep` client which is required for development'; \
+		exit 2; \
+	else \
+		dep version; \
+	fi
+
+get-dep:
+	# Install the latest release of dep
+	go get -d -u github.com/golang/dep
+	cd $(go env GOPATH)/src/github.com/golang/dep && \
+	DEP_TAG=$$(git describe --abbrev=0 --tags) && \
+	git checkout $$DEP_TAG && \
+	go install -ldflags="-X main.version=$$DEP_TAG" ./cmd/dep; \
+	git checkout master # Make go get happy by switching back to master
```

Comment:
Can we dep do the dep stuff in docker? fyi @duglin

Reply:
This needs to be removed (`get-dep` and `check-dep`) aren't used, because I baked dep into the docker image.

I originally had them there to help someone developing without docker, but because the Makefile redefines the GOPATH, it doesn't work anyway.